app/upgrades/v24/upgrades.go

```diff
@@ -41,6 +43,14 @@ func CreateUpgradeHandler(
 		// since we only need the pool indexed TWAPs.
 		keepers.TwapKeeper.DeleteAllHistoricalTimeIndexedTWAPs(ctx)
 
+		// restrict lockable durations to 2 weeks per
+		// https://wallet.keplr.app/chains/osmosis/proposals/400
+		keepers.IncentivesKeeper.SetLockableDurations(ctx, []time.Duration{
```

Comment:
Have we tested this by running through an epoch post-upgrade and confirming that incentives continue to distribute as intended?

Reply:
I have added a test here https://github.com/osmosis-labs/osmosis/pull/7677/commits/b981edb1111773e168d4eb973eed19ee7e2e7d48, please let me know if this suffices!